lib/ecto_ldap/adapter.ex

```diff
@@ -37,27 +38,48 @@ defmodule Ecto.Ldap.Adapter do
   # GenServer server API
   #
   ####
-  def handle_call({:search, search_options}, _from, state) do
+  def handle_call({:search, search_options, search_filter}, _from, state) do
+    {:ok, handle}   = ldap_connect(state)
+    search_response = ldap_search(handle, search_options, search_filter)
+    ldap_disconnect(handle)
+
+    {:reply, search_response, state}
+  end
+
+  def handle_call(:base, _from, state) do
+    base = Keyword.get(state, :base) |> to_char_list
+    {:reply, base, state}
+  end
 
+  def ldap_connect(state) do
     user_dn   = Keyword.get(state, :user_dn)  |> to_char_list
     password  = Keyword.get(state, :password) |> to_char_list
-    host      = Keyword.get(state, :host)   |> to_char_list
+    hostname  = Keyword.get(state, :hostname) |> to_char_list
```

Comment:
Gotcha.  Just wanted to ensure I called out the changes we had discussed, wasn't sure if GitHub gave enough insight to those changes.